spark/src/main/scala/org/apache/spark/sql/delta/commands/CreateDeltaTableCommand.scala

```diff
@@ -612,7 +612,7 @@ case class CreateDeltaTableCommand(
       deltaLog: DeltaLog,
       tableWithLocation: CatalogTable,
       snapshotOpt: Option[Snapshot] = None): OptimisticTransaction = {
-    val txn = deltaLog.startTransaction(snapshotOpt)
+    val txn = deltaLog.startTransaction(None, snapshotOpt)
```

Comment:
This could use a comment why we don't pass `tableWithLocation`. I believe the reason is, `tableWithLocation` is the _proposed_ catalog table entry, which will be installed by this command but does not yet exist as-of the start of the transaction?